custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -629,3 +662,180 @@ def get_queue(self):
             upload_burst_time, download_burst_time = self.data["queue"][uid]["burst-time"].split('/')
             self.data["queue"][uid]["upload-burst-time"] = upload_burst_time
             self.data["queue"][uid]["download-burst-time"] = download_burst_time
+
+    def build_accounting_hosts(self):
+        # Build hosts from DHCP Server leases and ARP list
+        self.data["accounting"] = parse_api(
+            data=self.data["accounting"],
+            source=self.api.path("/ip/dhcp-server/lease", return_list=True),
+            key="address",
```

Comment:
key should be mac-address, to retain history and custom names in HA ++ possibility for automation